### PR TITLE
AVX-97 create bootloader for BMU 4.3 hardware

### PR DESCRIPTION
**Pull request by rabad-matternet**
initial implementation of bootloader for BMU 4.3 hardware

### CODE REVIEW

File: rebuild_all.sh [hunk 1]
```diff
@@ -13,6 +13,7 @@ declare -a ALL_BOARDS=("com.matternet.carriage_2.0"
                        "com.matternet.ucann_1.0"
                        "com.matternet.bmu-B_1.0"
                        "com.matternet.external_led_1.0"
+                       "com.matternet.bmu_4.3"
```
Comment (zalice) [inline rebuild_all.sh:16]: Is BMU now major.minor.patch? should this be 4.3.0?

File: boards/com.matternet.bmu_4.3/board.ld [hunk 2]
```diff
@@ -0,0 +1,114 @@
+MEMORY
+{
+    bl (rx) :             ORIGIN = 0x08000000,            LENGTH = 12K
+    app (rx) :            ORIGIN = 0x08000000+12K,        LENGTH = 128K-12K-2K
+    params (rx) :         ORIGIN = 0x08000000+126K,        LENGTH = 2K
```
Comment (zalice) [inline boards/com.matternet.bmu_4.3/board.ld]: could you please align this LENGTH column with the others?

File: boards/com.matternet.bmu_4.3/board.ld [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+    } >PROGRAM_REGION
+    .fini_array : {
+        . = ALIGN(4);
+        __fini_array_start = .;
+        KEEP (*(.fini_array))
+        KEEP (*(SORT(.fini_array.*)))
+        __fini_array_end = .;
+    } >PROGRAM_REGION
+
+    /*
```
Comment (zalice) [inline boards/com.matternet.bmu_4.3/board.ld:68]: Can the `/*` and `*/` have consistent placement? in this block the `*/` is on a new line, in the comment block above, it's on the same line as the last sentence of the comment

File: boards/com.matternet.bmu_4.3/board.ld [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+        *(.bss*)	/* Read-write zero initialized data */
+        *(COMMON)
+        . = ALIGN(4);
+        _ebss = .;
+    } >ram
+
+    /*
+        * The .eh_frame section appears to be used for C++ exception handling.
+        * You may need to fix this if you're using C++.
+        */
```
Comment (zalice) [inline boards/com.matternet.bmu_4.3/board.ld]: Same comment here as above regarding the block comment start / end placement :)

**Review by zalice — APPROVED** [review]
A few nitpicks and comment about the version. Other than that 👍

File: rebuild_all.sh (hunk 1 above)
Comment (rabad-matternet) [reply rebuild_all.sh:16]: This is a hardware version.  Can we do patches on hardware?

Comment (zalice) [reply rebuild_all.sh:16]: Ah, I missed that this was hw version. The danger of hw version == sw version :)

File: boards/com.matternet.bmu_4.3/board.ld (hunk 2 above)
Comment (rabad-matternet) [reply boards/com.matternet.bmu_4.3/board.ld]: Done!

File: boards/com.matternet.bmu_4.3/board.ld (hunk 3 above)
Comment (rabad-matternet) [reply boards/com.matternet.bmu_4.3/board.ld:68]: Done!

File: boards/com.matternet.bmu_4.3/board.ld (hunk 4 above)
Comment (rabad-matternet) [reply boards/com.matternet.bmu_4.3/board.ld]: Done!

**Review by AydenChubbic — APPROVED** [review]

**Review by adriennet823 — APPROVED** [review]

**Pull request merged**